### PR TITLE
EVALSYS-1502: allow surveys to be transferred to a new owner

### PR DESCRIPTION
**Pull request by nicholaswilson100**

### CODE REVIEW

File: sakai-evaluation-impl/src/java/org/sakaiproject/evaluation/logic/EvalEvaluationServiceImpl.java [hunk 1]
```diff
@@ -15,8 +15,7 @@
 package org.sakaiproject.evaluation.logic;
 
 import java.io.OutputStream;
-import java.util.ArrayList;
-import java.util.Collection;
+import java.util.*;
```
Comment (bjones86) [inline sakai-evaluation-impl/src/java/org/sakaiproject/evaluation/logic/EvalEvaluationServiceImpl.java]: Please don't use * imports

File: sakai-evaluation-tool/src/java/org/sakaiproject/evaluation/tool/producers/ChownEvaluationProducer.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+	}
+
+	/* (non-Javadoc)
+	 * @see uk.org.ponder.rsf.viewstate.ViewParamsReporter#getViewParameters()
+	 */
+	public ViewParameters getViewParameters() {
+		return new EvalViewParameters();
+	}
+
+}
```
Comment (bjones86) [inline sakai-evaluation-tool/src/java/org/sakaiproject/evaluation/tool/producers/ChownEvaluationProducer.java]: Whitespace is very inconsistent in this file, and it has no EOF

File: sakai-evaluation-tool/src/java/org/sakaiproject/evaluation/tool/producers/ControlEvaluationsProducer.java [hunk 3]
```diff
@@ -202,11 +186,17 @@ public void fill(UIContainer tofill, ViewParameters viewparams, ComponentChecker
...
+            UIOutput.make(evaluationRow, "partial-eval-created", df.format(evaluation.getLastModified()));
+            EvalUser owner = commonLogic.getEvalUserById( evaluation.getOwner() );
+            UIOutput.make(evaluationRow, "partial-eval-owner", owner.displayName);
 
             UIInternalLink.make(evaluationRow, "partial-eval-edit-link", UIMessage.make("controlevaluations.partial.continue"),
                   new EvalViewParameters(EvaluationSettingsProducer.VIEW_ID, evaluation.getId()) );
-            UIInternalLink.make(evaluationRow, "partial-eval-delete-link", UIMessage.make("general.command.delete"), 
-                  new EvalViewParameters( RemoveEvalProducer.VIEW_ID, evaluation.getId() ) );
+
+             UIInternalLink.make(evaluationRow, "evaluation-chown-link", UIMessage.make("general.command.chown"),
```
Comment (bjones86) [inline sakai-evaluation-tool/src/java/org/sakaiproject/evaluation/tool/producers/ControlEvaluationsProducer.java]: Spacing looks a little different here than surrounding code

File: sakai-evaluation-tool/src/java/org/sakaiproject/evaluation/tool/producers/ControlEvaluationsProducer.java [hunk 4]
```diff
@@ -256,6 +246,9 @@ public void fill(UIContainer tofill, ViewParameters viewparams, ComponentChecker
             UIInternalLink.make(evaluationRow, "inqueue-eval-edit-link", UIMessage.make("general.command.edit"),
                   new EvalViewParameters(EvaluationSettingsProducer.VIEW_ID, evaluation.getId()) );
 
+             UIInternalLink.make(evaluationRow, "evaluation-chown-link", UIMessage.make("general.command.chown"),
```
Comment (bjones86) [inline sakai-evaluation-tool/src/java/org/sakaiproject/evaluation/tool/producers/ControlEvaluationsProducer.java]: And here too

File: sakai-evaluation-tool/src/java/org/sakaiproject/evaluation/tool/producers/ControlEvaluationsProducer.java [hunk 5]
```diff
@@ -343,6 +338,8 @@ public void fill(UIContainer tofill, ViewParameters viewparams, ComponentChecker
 
             humanDateRenderer.renderDate(evaluationRow, "active-eval-startdate", evaluation.getStartDate());
             // TODO add support for evals that do not close - summary.label.nevercloses
+             EvalUser owner = commonLogic.getEvalUserById( evaluation.getOwner() );
```
Comment (bjones86) [inline sakai-evaluation-tool/src/java/org/sakaiproject/evaluation/tool/producers/ControlEvaluationsProducer.java]: And here.

File: sakai-evaluation-tool/src/java/org/sakaiproject/evaluation/tool/producers/ControlEvaluationsProducer.java [hunk 6]
```diff
@@ -397,7 +394,10 @@ public void fill(UIContainer tofill, ViewParameters viewparams, ComponentChecker
 
             UIInternalLink.make(evaluationRow, "closed-eval-notifications-link", UIMessage.make("controlevaluations.eval.email.link"),
                     new EvalViewParameters( EvaluationNotificationsProducer.VIEW_ID, evaluation.getId() ) );
-
+             EvalUser owner = commonLogic.getEvalUserById( evaluation.getOwner() );
```
Comment (bjones86) [inline sakai-evaluation-tool/src/java/org/sakaiproject/evaluation/tool/producers/ControlEvaluationsProducer.java]: And here

File: sakai-evaluation-tool/src/webapp/WEB-INF/requestContext.xml [hunk 7]
```diff
@@ -544,6 +544,12 @@
 		<property name="navBarRenderer" ref="navBarRenderer" />	
     </bean>
 
+	<bean class="org.sakaiproject.evaluation.tool.producers.ChownEvaluationProducer">
```
Comment (bjones86) [inline sakai-evaluation-tool/src/webapp/WEB-INF/requestContext.xml]: Whitespace...

File: sakai-evaluation-tool/src/webapp/content/templates/chown_evaluation.html [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+	<link href="../css/evaluation_base.css" type="text/css" rel="stylesheet" media="all" />
+</head>
+
+<body>
+<div class="portletBody">
+
+	<div rsf:id="navIntraTool:" class="navIntraTool"/>
+
+	<ul class="breadCrumb">
+		<li><a rsf:id="summary-link" href="summary.html">Summary</a></li><li><a rsf:id="control-evaluations-link" href="evaluations.html">My Evaluations</a></li><li class="lastCrumb" rsf:id="chown-evaluation-title">Change evaluation Owner</li>
```
Comment (bjones86) [inline sakai-evaluation-tool/src/webapp/content/templates/chown_evaluation.html]: Can you put these on separate lines?

File: sakai-evaluation-tool/src/webapp/content/templates/chown_evaluation.html [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+					<input name="chown" rsf:id="chown-evaluation-newowner"   id="chown"  type="text" value="" size="50" maxlength="255" />
+				</p>
+				<input rsf:id="chown-evaluation-button" type="submit" value="Chown Evaluation" class="active" accesskey="s" />
+			</form>
+			<input rsf:id="cancel-command-link" type="button" value="Cancel" onclick="javascript:history.go(-1);return false;" accesskey="x"/>
+		</div>
+	</div>
+</div>
+</body>
+</html>
```
Comment (bjones86) [inline sakai-evaluation-tool/src/webapp/content/templates/chown_evaluation.html]: EOF

File: sakai-evaluation-tool/src/webapp/content/templates/control_evaluations.html [hunk 10]
```diff
@@ -72,6 +72,9 @@ <h3 rsf:id="msg=controlevaluations.partial.header">Partial Evaluations</h3>
                         <th rsf:id="msg=controlevaluations.partial.created.date.header" width="10%">
                             Created On
                         </th>
+                        <th rsf:id="msg=controlevaluations.partial.owner.header" width="10%">
+                            Owner
+                         </th>
```
Comment (bjones86) [inline sakai-evaluation-tool/src/webapp/content/templates/control_evaluations.html]: Whitespace

File: sakai-evaluation-tool/src/webapp/content/templates/control_evaluations.html [hunk 11]
```diff
@@ -138,6 +149,9 @@ <h3 rsf:id="msg=controlevaluations.inqueue.header">In-Queue Evaluations</h3>
                         <td rsf:id="inqueue-eval-duedate" nowrap="nowrap" headers="dueinqueue">
                             20/03/2007
                         </td>
+                        <td rsf:id="inqueue-eval-owner" nowrap="nowrap" headers="ownerinqueue">
+                            Owner
+                         </td>
```
Comment (bjones86) [inline sakai-evaluation-tool/src/webapp/content/templates/control_evaluations.html]: Whitespace

File: sakai-evaluation-tool/src/webapp/content/templates/control_evaluations.html [hunk 12]
```diff
@@ -175,6 +189,9 @@ <h3 rsf:id="msg=controlevaluations.active.header" style="font-weight:bold;">Acti
                         <th rsf:id="msg=controlevaluations.eval.report.header" width="15%" class="screenOnly">
                             Report
                         </th>
+                        <th rsf:id="msg=controlevaluations.eval.owner.header" width="10%">
+                            Owner
+                         </th>
```
Comment (bjones86) [inline sakai-evaluation-tool/src/webapp/content/templates/control_evaluations.html]: Whitespace

**Comment by jonespm** [comment]
@nicholaswilson100 Do you have any time to fix up these issues mentioned here?

**Comment by nicholaswilson100** [comment]
Yes I'll try and do it soon

**Comment by nicholaswilson100** [comment]
I've fixed the issues  - couldn't seem to fix the EOF in chown_evaluation.html. Can't see why.

**Comment by jonespm** [comment]
Okay, thanks I'll merge this would be nice if it could get verified too.

**Pull request merged**